pkg/proxy/ipvs/proxier.go

```diff
@@ -1982,6 +1998,15 @@ func (proxier *Proxier) syncEndpoint(svcPortName proxy.ServicePortName, onlyNode
 	for _, epInfo := range endpoints {
 		newEndpoints.Insert(epInfo.String())
 	}
+	if svcInfo.UsesLocalEndpoints() && len(localEndpoints) == 0 {
```

Comment:
len(localEndpoints) == 0 can be replaced with len(endpoints) == 0, since endpoints was set to localEndpoitns on line 1987, which also removes the extra variable declaration added on line 1975 and line 1984

Reply:
I would also replace `svcInfo.UsesLocalEndpoints()` check here with just `onlyNodeLocalEndpoints`, because the type of VIP matters for IPVS compared to iptables and `onlyNodeLocalEndpoints` will explicitly be set 
```
if onlyNodeLocalEndpoints && len(endpoints) == 0 {
    ...
}
```

Reply:
That's how I implemented it first but onlyNodeLocalEndpoints is false when the external traffic policy is set to local.
That also means that endpoints will be the number of clusterEndpoints.
`$ go test . -run TestNoEndpointsMetric
Testing scenario: internalTrafficPolicy is set and there is non-zero local endpoints
onlyNodeLocalEndpoints == true, svcInfo.UsesLocalEndpoints() == true, len(endpoints) == 1
Testing scenario: externalTrafficPolicy is set and there is non-zero local endpoints
onlyNodeLocalEndpoints == false, svcInfo.UsesLocalEndpoints() == true, len(endpoints) == 3
Testing scenario: internalTrafficPolicy is set and there is zero local endpoint
onlyNodeLocalEndpoints == true, svcInfo.UsesLocalEndpoints() == true, len(endpoints) == 0
Testing scenario: externalTrafficPolicy is set and there is zero local endpoint
onlyNodeLocalEndpoints == false, svcInfo.UsesLocalEndpoints() == true, len(endpoints) == 3
--- FAIL: TestNoEndpointsMetric (0.00s)
    proxier_test.go:5584: sync_proxy_rules_no_endpoints_total metric mismatch(internal): got=0, expected 1
FAIL
FAIL    k8s.io/kubernetes/pkg/proxy/ipvs        0.142s
FAIL`